### PR TITLE
api, rest: enable filtering specific fields

### LINKED ISSUE #903: Make possible to pick api object fields
Currently when retrieving objects from the API, squad returns the full attribute list of objects, it would be nice to select which fields to fetch, in case just a few is needed.

The idea is based on this: https://stackoverflow.com/a/23674297

It'll help squad-client fetch only wanted fields https://github.com/Linaro/squad-client/issues/72

### PR DESCRIPTION
**Pull request by amrohassaan**
Fixes #903. Enable filtering specific fields for queried objects.

### CODE REVIEW

**Comment by chaws** [comment]
Awesome, could you write some tests for it? just as a poc

**Comment by amrohassaan** [comment]
> Awesome, could you write some tests for it? just as a poc

Ack @chaws

**Comment by mwasilew** [comment]
Looks OK. Please also add a paragraph in the docs on how to use it. It's obvious from the code, but not everyone reads the code :P

**Comment by amrohassaan** [comment]
@chaws tests added. @mwasilew ack

File: test/api/test_rest.py [hunk 1]
```diff
@@ -810,6 +814,11 @@ def test_groups(self):
         data = self.hit('/api/groups/')
         self.assertEqual('mygroup', data['results'][0]['slug'])
 
+    def test_groups_slug_field_only(self):
+        data = self.hit('/api/groups/?fields=slug')
```
Comment (mwasilew) [inline test/api/test_rest.py:818]: how about a test with more than 1 field?

Comment (amrohassaan) [reply test/api/test_rest.py:818]: @mwasilew wouldn't harm at all :P

Comment (amrohassaan) [reply test/api/test_rest.py:823]: @mwasilew add the test requested.

**Comment by chaws** [comment]
LGTM

**Pull request merged**